packages/truffle-config/index.js

```diff
@@ -344,14 +344,17 @@ Config.detect = function(options, filename) {
 };
 
 Config.load = function(file, options) {
-  var config = new Config();
+  var static_config = originalrequire(file);
+
+  var config = static_config.config
+    ? new static_config.config()
+    : new Config();
 
   config.working_directory = path.dirname(path.resolve(file));
 
   // The require-nocache module used to do this for us, but
   // it doesn't bundle very well. So we've pulled it out ourselves.
   delete require.cache[Module._resolveFilename(file, module)];
-  var static_config = originalrequire(file);
```

Comment:
We need to be sure that the `delete` happening before the `originalrequire` is not necessary. Might be safer just to do the loading after the `delete`, like it is currently.